typegate/src/engine.ts

```diff
@@ -40,54 +34,24 @@ const introspectionDefStatic = await Deno.readTextFile(
   join(localDir, "typegraphs/introspection.json"),
 );
 
-const typegraphChangelog: Record<
-  string,
-  { next: string; transform: (x: TypeGraphDS) => TypeGraphDS }
-> = {
-  "0.0.0": {
-    "next": "0.0.1",
-    "transform": (x) => x,
-  },
-};
-
-function upgradeTypegraph(typegraph: TypeGraphDS) {
-  const typegraphName = typegraph.types[0].title;
-  const { meta } = typegraph;
-
-  let currentVersion = meta.version;
-  while (semver.neq(typegraphVersion, currentVersion)) {
-    const migration = typegraphChangelog[currentVersion];
-    if (!migration) {
-      throw Error(
-        `typegate ${config.version} supports typegraph ${typegraphVersion} which is incompatible with ${typegraphName} ${meta.version} (max auto upgrade was ${currentVersion})`,
-      );
-    }
-    typegraph = migration.transform(typegraph);
-    currentVersion = migration.next;
-  }
-
-  return typegraph;
-}
-
 export const initTypegraph = async (
   payload: string,
+  sync: boolean, // redis synchronization?
```

Comment:
maybe choose a better name, e.g. "nopushhook" as it does not seem related to the sync?

Reply:
It indicates if the typegraph is being initialized from the redis synchronization or a direct push from the user...

Reply:
"onPush" hooks will only run on one instance of the typegate: the one that received the typegraph directly from the user. The others (getting their versions from redis) will not run the hooks...